Source/SwiftLintCore/Protocols/SwiftSyntaxRule.swift

```diff
@@ -127,3 +127,14 @@ public extension Array where Element == ReasonedRuleViolation {
         append(contentsOf: positions.map { ReasonedRuleViolation(position: $0) })
     }
 }
+
+/// Extension for sets of `ReasonedRuleViolation`s that provides the automatic conversion of
+/// `AbsolutePosition`s into `ReasonedRuleViolation`s (without a specific reason).
+public extension Set where Element == ReasonedRuleViolation {
```

Comment:
it took me a while to work out how `AbsolutePosition`'s got "magically" transformed into `ReasonedRuleViolations` when they got added to `violations`.

Reply:
Well, is that really needed? You can carry around just the positions in `possibleViolations` and when they are eventually transferred into the `violations` array they will be converted to `ReasonedRuleViolations` as usual.

I think we should avoid the public extension if possible.

Reply:
Cool - I'll rework that later today or tomorrow

Reply:
ok - should be good now I think ... I also converted the Examples to string literals, a la #5154

Reply:
The string literals seemed to be cool. But unfortunately they break reporting of test failures right at the example code. I'll have to revert the change until we know how to make this work.

Reply:
Dammit - reverted mine for now.